array_family/flex_ext.py

```diff
@@ -1239,10 +1239,10 @@ def map_centroids_to_reciprocal_space(
         panel_numbers = cctbx.array_family.flex.size_t(self["panel"])
 
         for i, expt in enumerate(experiments):
-            if "imageset_id" in self:
-                sel_expt = self["imageset_id"] == i
-            else:
+            if "id" in self:
                 sel_expt = self["id"] == i
+            else:
+                sel_expt = self["imageset_id"] == i
```

Comment:
Have you ran the indexing tests after making this change? I'm pretty sure this will result in some indexing failures as it explicitly relies on the existing behaviour - i.e. use the `imageset_id`s which are distinct, rather than the `id`s which will have been reset to `-1` at the beginning of indexing to indicate "unindexed". (Yes, I know #1029 #1104.)

Reply:
Of course: this is precisely why I flagged this very early as a draft PR as I thought this will be über destructive - though that to me indicates that we have a _different_ issue... but point well made.

Reply:
Unsurprisingly this breaks many things

```
Results (264.31s):
    1022 passed
       4 failed
         - tests/algorithms/indexing/test_index.py:753 test_index_known_orientation
         - tests/algorithms/indexing/test_assign_indices.py:247 test_local_multiple_rotations
         - tests/command_line/test_export_best.py:4 test_export_best
         - tests/command_line/test_search_beam_position.py:170 test_search_small_molecule
     157 skipped
```